src/EditorFeatures/Core/Navigation/AbstractDefinitionLocationService.cs

```diff
@@ -39,47 +39,56 @@ internal abstract class AbstractDefinitionLocationService : IDefinitionLocationS
             workspace, document.Id, position, virtualSpace: 0, cancellationToken);
     }
 
-    public async Task<DefinitionLocation?> GetDefinitionLocationAsync(
-        Document document, int position, CancellationToken cancellationToken)
+    public async Task<DefinitionLocation?> GetDefinitionLocationAsync(Document document, int position, CancellationToken cancellationToken)
     {
-        var symbolService = document.GetRequiredLanguageService<IGoToDefinitionSymbolService>();
-        var (controlFlowTarget, controlFlowSpan) = await symbolService.GetTargetIfControlFlowAsync(
-            document, position, cancellationToken).ConfigureAwait(false);
-        if (controlFlowTarget != null)
-        {
-            var location = await GetNavigableLocationAsync(
-                document, controlFlowTarget.Value, cancellationToken).ConfigureAwait(false);
-            return location is null ? null : new DefinitionLocation(location, new DocumentSpan(document, controlFlowSpan));
-        }
-        else
+        // We want to compute this as quickly as possible so that the symbol be squiggled and navigated to.  We
+        // don't want to wait on expensive operations like computing source-generators or skeletons if we can avoid
+        // it.  So first try with a frozen document, then fallback to a normal document.  This mirrors how go-to-def
+        // works as well.
+        return await GetDefinitionLocationWorkerAsync(document.WithFrozenPartialSemantics(cancellationToken)).ConfigureAwait(false) ??
```

Comment:
this is what we did in https://github.com/dotnet/roslyn/pull/72714.  however, we still have not *fully* unified all the codepaths involved in navigating to a symbol :'(   so there's still the need to update in a one-off manner like this.  ideally this can be unified in the future.